### ISSUE TITLE
Error creating controller. Please check Docker is running using `docker ps`.

### ISSUE THREAD

**Issue by tonyw54** [OpenDevin/OpenDevin#1078]
I'm running WSL, Ubuntu on a Microsoft surface.

I pulled from main and ran these:

make build
make run

I then went to port 3001 in chrome.

The backend and front end started, but I'm getting errors in the chat console saying it could not create the controller.

![Screenshot 2024-04-13 131757](https://github.com/OpenDevin/OpenDevin/assets/36421300/e60747a9-37a6-4543-9832-5e91fa743ae8)

I removed all my images from docker and did another build and run, but the problem still persists.

#### Setup and configuration

**Current version**:
Result of running <git log -n 1>

commit 8b9f13b1ed4949bafc6dcfbe66d405ce5b4d7e3b (HEAD -> main, origin/main, origin/HEAD)
Author: Akki <akhilvc10@gmail.com>
Date:   Sat Apr 13 19:52:26 2024 +0400

    fix(editor): ui enhancements and code refactor (#1069)

```bash
```

**My config.toml and environment vars** (be sure to redact API keys):
```toml

LLM_API_KEY="redacted"
LLM_MODEL="gpt-3.5-turbo"
WORKSPACE_DIR="./workspace"

```

**My model and agent** (you can see these settings in the UI):
* Model: gpt-3.5-turbo
* Agent: Monologueagent

**Commands I ran to install and run OpenDevin**:
make build
make run

**Steps to Reproduce**:
1. pull latest repo from git: "git pull origin main"
2. make build
3. make run
4. Go to http://localhost:3001/ in google chrome.

**Logs, error messages, and screenshots**:

Here is the console output of my run. This includes the "make run" command, then going to port 3001 in Chrome:

~/OpenDevin$ make run
Running the app...
Starting backend server...
Waiting for the backend to start...
/home/tonyw/.cache/pypoetry/virtualenvs/opendevin-HG9b_BmJ-py3.11/lib/python3.11/site-packages/pydantic/_internal/_fields.py:151: UserWarning: Field "model_list" has conflict with protected namespace "model_".

You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
  warnings.warn(
/home/tonyw/.cache/pypoetry/virtualenvs/opendevin-HG9b_BmJ-py3.11/lib/python3.11/site-packages/pydantic/_internal/_fields.py:151: UserWarning: Field "model_name" has conflict with protected namespace "model_".

You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
  warnings.warn(
/home/tonyw/.cache/pypoetry/virtualenvs/opendevin-HG9b_BmJ-py3.11/lib/python3.11/site-packages/pydantic/_internal/_fields.py:151: UserWarning: Field "model_group_alias" has conflict with protected namespace "model_".

You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
  warnings.warn(
/home/tonyw/.cache/pypoetry/virtualenvs/opendevin-HG9b_BmJ-py3.11/lib/python3.11/site-packages/pydantic/_internal/_fields.py:151: UserWarning: Field "model_info" has conflict with protected namespace "model_".

You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
  warnings.warn(
/home/tonyw/.cache/pypoetry/virtualenvs/opendevin-HG9b_BmJ-py3.11/lib/python3.11/site-packages/pydantic/_internal/_fields.py:151: UserWarning: Field "model_id" has conflict with protected namespace "model_".

You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
  warnings.warn(
INFO:     Started server process [7912]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:3000 (Press CTRL+C to quit)
Backend started successfully.
Starting frontend with npm...

> opendevin-frontend@0.1.0 start
> vite --port 3001

  VITE v5.2.8  ready in 635 ms

  ➜  Local:   http://localhost:3001/
  ➜  Network: use --host to expose
  ➜  press h + enter to show help
INFO:     127.0.0.1:47238 - "GET /refresh-files HTTP/1.1" 200 OK
INFO:     127.0.0.1:47242 - "GET /configurations HTTP/1.1" 200 OK
INFO:     127.0.0.1:47244 - "GET /messages/total HTTP/1.1" 200 OK
INFO:     127.0.0.1:47254 - "GET /refresh-files HTTP/1.1" 200 OK
INFO:     ('127.0.0.1', 47256) - "WebSocket /ws?token=***" [accepted]
Starting loop_recv for sid: 72f7fa64-b3f1-49f0-920d-d8d9914c85b0
INFO:     connection open
INFO:     127.0.0.1:47270 - "GET /messages/total HTTP/1.1" 200 OK
INFO:     127.0.0.1:47276 - "GET /configurations HTTP/1.1" 200 OK
13:23:55 - opendevin:INFO: ssh_box.py:80 - Using workspace directory: /home/tonyw/OpenDevin/workspace
13:23:56 - opendevin:INFO: ssh_box.py:280 - Container stopped
13:23:56 - opendevin:INFO: ssh_box.py:312 - Container started
13:23:57 - opendevin:INFO: ssh_box.py:329 - waiting for container to start: 1, container status: running
13:23:58 - opendevin:INFO: ssh_box.py:169 - Connecting to opendevin@localhost via ssh. If you encounter any issues, you can try `ssh -v -p 50913 opendevin@localhost` with the password '***' and report the issue on GitHub.
13:23:58 - opendevin:ERROR: agent.py:136 - Error creating controller: Could not establish connection to host
Traceback (most recent call last):
  File "/home/tonyw/OpenDevin/opendevin/server/agent/agent.py", line 126, in create_controller
    self.controller = AgentController(
                      ^^^^^^^^^^^^^^^^
  File "/home/tonyw/OpenDevin/opendevin/controller/agent_controller.py", line 92, in __init__
    self.command_manager = CommandManager(
                           ^^^^^^^^^^^^^^^
  File "/home/tonyw/OpenDevin/opendevin/controller/command_manager.py", line 26, in __init__
    self.shell = DockerSSHBox(
                 ^^^^^^^^^^^^^
  File "/home/tonyw/OpenDevin/opendevin/sandbox/ssh_box.py", line 98, in __init__
    self.start_ssh_session()
  File "/home/tonyw/OpenDevin/opendevin/sandbox/ssh_box.py", line 174, in start_ssh_session
    self.ssh.login(hostname, username, self._ssh_password, port=self._ssh_port)
  File "/home/tonyw/.cache/pypoetry/virtualenvs/opendevin-HG9b_BmJ-py3.11/lib/python3.11/site-packages/pexpect/pxssh.py", line 424, in login
    raise ExceptionPxssh('Could not establish connection to host')
pexpect.pxssh.ExceptionPxssh: Could not establish connection to host
13:23:58 - opendevin:INFO: ssh_box.py:80 - Using workspace directory: /home/tonyw/OpenDevin/workspace
13:23:59 - opendevin:INFO: ssh_box.py:280 - Container stopped
13:23:59 - opendevin:INFO: ssh_box.py:312 - Container started
13:24:00 - opendevin:INFO: ssh_box.py:329 - waiting for container to start: 1, container status: running
13:24:01 - opendevin:INFO: ssh_box.py:169 - Connecting to opendevin@localhost via ssh. If you encounter any issues, you can try `ssh -v -p 56717 opendevin@localhost` with the password '***' and report the issue on GitHub.
13:24:01 - opendevin:ERROR: agent.py:136 - Error creating controller: Could not establish connection to host
Traceback (most recent call last):
  File "/home/tonyw/OpenDevin/opendevin/server/agent/agent.py", line 126, in create_controller
    self.controller = AgentController(
                      ^^^^^^^^^^^^^^^^
  File "/home/tonyw/OpenDevin/opendevin/controller/agent_controller.py", line 92, in __init__
    self.command_manager = CommandManager(
                           ^^^^^^^^^^^^^^^
  File "/home/tonyw/OpenDevin/opendevin/controller/command_manager.py", line 26, in __init__
    self.shell = DockerSSHBox(
                 ^^^^^^^^^^^^^
  File "/home/tonyw/OpenDevin/opendevin/sandbox/ssh_box.py", line 98, in __init__
    self.start_ssh_session()
  File "/home/tonyw/OpenDevin/opendevin/sandbox/ssh_box.py", line 174, in start_ssh_session
    self.ssh.login(hostname, username, self._ssh_password, port=self._ssh_port)
  File "/home/tonyw/.cache/pypoetry/virtualenvs/opendevin-HG9b_BmJ-py3.11/lib/python3.11/site-packages/pexpect/pxssh.py", line 424, in login
    raise ExceptionPxssh('Could not establish connection to host')
pexpect.pxssh.ExceptionPxssh: Could not establish connection to host
INFO:     127.0.0.1:43114 - "GET /litellm-models HTTP/1.1" 200 OK
INFO:     127.0.0.1:43128 - "GET /litellm-agents HTTP/1.1" 200 OK
INFO:     127.0.0.1:43140 - "GET /litellm-models HTTP/1.1" 200 OK
INFO:     127.0.0.1:43146 - "GET /litellm-agents HTTP/1.1" 200 OK

**Comment by enyst** [comment]
Can you add to config.toml : SANDBOX_TYPE = exec, then run it again?

**Comment by tonyw54** [comment]
Yep, I modified my config.toml to this and it worked. Can you explain why that fixed it?

LLM_API_KEY="<redacted>"
LLM_MODEL="gpt-3.5-turbo"
WORKSPACE_DIR="./workspace"
SANDBOX_TYPE="exec"

**Comment by rbren** [comment]
`SANDBOX_TYPE="exec"` is a temporary workaround. We're trying to get rid of the underlying problem

**Comment by tonyw54** [comment]
A workaround has been found for this issue, for now

**Comment by hongthana** [comment]
I have the same issue. I have added `SANDBOX_TYPE="exec"` in the config.toml. I am using Mac and Docker desktop. I checked the docker `docker ps`. There is nothing running. I checked inside docker desktop. The image is there but not running. When I manually clicked run from the docker desktop, I got `Exited (0)`

`config.toml`
LLM_MODEL="claude-3-opus-20240229"
LLM_API_KEY=""
LLM_EMBEDDING_MODEL=""
WORKSPACE_DIR="./workspace"
SANDBOX_TYPE="exec"

Here is the log:
INFO:     Started server process [7409]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:3000 (Press CTRL+C to quit)
INFO:     ('127.0.0.1', 53398) - "WebSocket /ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzaWQiOiIzOTU4MWZkOC0zYWQ2LTQwNTYtYTRmMi1hNGZjNzQyMWZhODQifQ.cfABOSoUuWHjFJjk82AhH51S5dxgZL66I0X_wHX-oTs" [accepted]
INFO:     connection open
Starting loop_recv for sid: 39581fd8-3ad6-4056-a4f2-a4fc7421fa84, False
Error creating controller.
INFO:     127.0.0.1:53401 - "GET /litellm-models HTTP/1.1" 200 OK
INFO:     127.0.0.1:53402 - "GET /litellm-agents HTTP/1.1" 200 OK
INFO:     127.0.0.1:53403 - "GET /messages/total HTTP/1.1" 200 OK
INFO:     127.0.0.1:53399 - "GET /refresh-files HTTP/1.1" 200 OK
INFO:     127.0.0.1:53404 - "GET /litellm-models HTTP/1.1" 200 OK
INFO:     127.0.0.1:53400 - "GET /default-model HTTP/1.1" 200 OK
INFO:     127.0.0.1:53406 - "GET /litellm-agents HTTP/1.1" 200 OK
INFO:     127.0.0.1:53410 - "GET /messages/total HTTP/1.1" 200 OK
INFO:     127.0.0.1:53412 - "GET /refresh-files HTTP/1.1" 200 OK
INFO:     127.0.0.1:53413 - "GET /default-model HTTP/1.1" 200 OK
INFO:     127.0.0.1:53419 - "GET /messages HTTP/1.1" 200 OK
